### PR TITLE
added lspci module

### PR DESCRIPTION
**Pull request by amitsagtani97**

### CODE REVIEW

File: transcribe/scribe_modules/lspci.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if input_dict:
+            self.value = self._parse(input_dict)
+
+    def __iter__(self):
+        for attr, value in self.__dict__.items():
+            yield attr, value
+
+    def _parse(self, lspci_data):
+        output = []
+        lspci_lines = lspci_data.get("stockpile_lspci").split('\n\n')
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py]: We shouldn't be needing to do this..

File: transcribe/scribe_modules/lspci.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        if len(lspci_lines) <= 1:
+            print("Error occured in processing sysctl data")
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py]: this is lspci you mean ? Also maybe update the error to data not present?

File: transcribe/scribe_modules/lspci.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            sys.exit(1)
+
+        for i in range(len(lspci_lines)):
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py:26]: Add a small detailed explanation on why you're building this dictionary this way ?

File: transcribe/scribe_modules/lspci.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        # We are yielding a dict of these individual slots.
+        for i in range(len(lspci_lines)):
+            slot_data = lspci_lines[i].split('\n')
+            # sample - ['Slot:\t00:00.0', 'Class:\tHost bridge', 'Vendor:\tIntel Corporation' ..... ]
+            slot_dict = {}
+            for slot_info in slot_data:
+                key = slot_info.split(':\t')[0].strip()
+                value = slot_info.split(':\t')[1].strip()
+                slot_dict[key] = value
+            yield slot_dict
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py]: please yield self._dict or a copy of it

File: transcribe/scribe_modules/lspci.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+    def __iter__(self):
+        for attr, value in self.__dict__.items():
+            yield attr, value
+
+    def parse(self):
+        lspci_data = self._input_dict
+        lspci_lines = lspci_data.split('\n\n')
+
+        if len(lspci_lines) <= 1:
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py]: please dont use this, use the function call ` validate_length(len(lspci_lines), self.module)`

File: transcribe/scribe_modules/lspci.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    def __init__(self, input_dict=None, module_name=None, host_name=None,
+                 input_type=None, scribe_uuid=None):
+        ScribeModuleBaseClass.__init__(self, module_name=module_name,
+                                       input_dict=input_dict,
+                                       host_name=host_name,
+                                       input_type=input_type,
+                                       scribe_uuid=scribe_uuid)
+
+    def __iter__(self):
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py]: This is not needed anymore

File: transcribe/schema/lspci.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+scribe_uuid:
```
Comment (aakarshg) [inline transcribe/schema/lspci.yml:1]: this is missing the following keys:
```
1. 'Slot'
2. 'Class'
3. 'Vendor'
4. 'Device'
5. 'SVendor'
6. 'SDevice'
7. 'Rev'
8. 'Driver'
```

File: transcribe/scribe_modules/lspci.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        lspci_lines = lspci_data.split('\n\n')
+
+        validate_length(len(lspci_lines), self.module)
+
+        # lspci_lines contains a list of slots and its information.
+        # We are yielding a dict of these individual slots.
+        for i in range(len(lspci_lines)):
+            slot_data = lspci_lines[i].split('\n')
+            # sample - ['Slot:\t00:00.0', 'Class:\tHost bridge', 'Vendor:\tIntel Corporation' ..... ]
+            self._dict["value"] = {}
```
Comment (aakarshg) [inline transcribe/scribe_modules/lspci.py:29]: move this to line 31, so that we make sure we're not reusing any key,value pairs from previous slot data

Comment (amitsagtani97) [reply transcribe/scribe_modules/lspci.py:29]: The values are already reset for each slot data in the first for loop. The second loop is for data in each slot, so we will need to move this.

**Review by dry923 — APPROVED** [review]
LGTM

**Pull request merged**